docs/source/configuration/entity-caching.mdx

```diff
@@ -0,0 +1,126 @@
+- Share the product cache between all users.
+- Cache the cart per user, with a small amount of data.
+- Cache inventory data with a low TTL or not cache it at all.
+
+## Use entity caching
+
+Follow this guide to enable and configure entity caching in the Apollo Router.
+
+### Prerequisites
+
+To use entity caching in the Apollo Router, you must set up:
+
+- A Redis instance or cluster that your router instances can communicate with.
+- A [GraphOS Enterprise plan](https://www.apollographql.com/pricing/) that [connects your router to GraphOS](./overview/#environment-variables).
+
+### Configure router for entity caching
+
+In `router.yaml`, configure `experimental_entity_cache`:
+- Enable entity caching globally.
+- Configure Redis, using the same conventions described in [distributed caching](./distributed-caching).
+- Configure entity caching per subgraph, with overrides per subgraph for disabling entity caching and TTL.
+
+For example:
+
+```yaml title="router.yaml"
+experimental_entity_cache:
+  enabled: true
+
+  redis:
+    urls: ["redis://..."]
+    timeout: 5ms # Optional, by default: 2ms
+    ttl: 24h # Optional, by default no expiration
+
+  subgraphs:
+    accounts:
+      enabled: false # disable for a specific subgraph
+    reviews:
+      ttl: 120s # overrides the global TTL
+```
+
+### Configure time to live (TTL)
+
+Besides configuring a global TTL for all the entries in Redis, the Apollo Router also honors the `Cache-Control` header returned with the subgraph response. It generates a `Cache-Control` header for the client response by aggregating the TTL information from all response parts.
+
+### Customize Redis cache key
+
+If you need to store data for a particular query in different cache entries, the cache key can be configured through the `apollo_entity_cache::key` context entry. This entry contains an object with the `all` field to affect all subgraph queries under one client query, and fields named after subgraph operation names to affect individual subgraph queries. The value of the field can be any valid JSON value (object, string, etc).
+
+```json
+{
+    "all": 1,
+    "subgraph_operation1": "key1",
+    "subgraph_operation2": {
+      "data": "key2"
+    }
+}
+
+```
+
+## Implementation notes
+
+### Responses with errors not cached
+
+To prevent transient errors from affecting the cache for a long duration, subgraph responses with errors are not cached.
+
+### Authorization and entity caching
+
+When used alongside the router's [authorization directives](./authorization), cache entries are separated by authorization context. If a query contains fields that need a specific scope, the requests providing and not providing that scope have different cache entries. This means that data requiring authorization can still be safely cached and even shared across users, without needing invalidation when a user's roles change because their queries are automatically directed to a different part of the cache.
```

Comment:
```suggestion
When used alongside the router's [authorization directives](./authorization), cache entries are separated by authorization context. If a query contains fields that need a specific scope, the requests providing have different cache entries from those not providing that scope. This means that data requiring authorization can still be safely cached and even shared across users, without needing invalidation when a user's roles change because their requests are automatically directed to a different part of the cache.
```

This second sentence could still use some breaking up into shorter pieces.